backend/src/test/kotlin/com/group7/artshare/controller/UserInteractionControllerTest.kt

```diff
@@ -0,0 +1,89 @@
+package com.group7.artshare.controller
+
+import com.group7.artshare.entity.*
+import com.group7.artshare.service.JwtService
+import com.group7.artshare.service.ProfileService
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito.anyString
+import org.mockito.Mockito.`when`
+import org.mockito.junit.jupiter.MockitoExtension
+import org.springframework.http.HttpStatus
+import org.springframework.web.server.ResponseStatusException
+import java.util.*
+
+
+@ExtendWith(MockitoExtension::class)
+internal class UserInteractionControllerTest {
+
+    @InjectMocks
+    lateinit var userInteractionController: UserInteractionController
+
+    @Mock
+    lateinit var profileService: ProfileService
+
+    @Mock
+    lateinit var jwtService: JwtService
+
+    @Test
+    fun successfullyFollowsUser() {
+        val accountInfo1 = AccountInfo("email@email.com", "janedoe", "31415926")
+        val mockUser = RegisteredUser(accountInfo1, setOf())
+        `when`(profileService.followUser("username", mockUser)).thenReturn(HttpStatus.ACCEPTED)
+        `when`(jwtService.getUserFromAuthorizationHeader("authorizationHeader")).thenReturn(mockUser)
+        val response = userInteractionController.followUser("username", "authorizationHeader")
+        assert(response == Unit)
+    }
+
+    @Test
+    fun failsIfFollowItself() {
+        val accountInfo1 = AccountInfo("email@email.com", "janedoe", "31415926")
+        val mockUser = RegisteredUser(accountInfo1, setOf())
+        `when`(
+            profileService.followUser(
+                "username",
```

Comment:
Shouldn't this "username" be given janedoe if you want to test the scenario in the function name. Also this would require anyString() in line 54 to be "janedoe". And also this would cause the second "when" starting in line 51 to be deleted completely because that is what you are testing if will happen in assertThrows, you shouldn't enforce it like this. I wrote it as below (sharing because the feedback is complicated written like above) if you would consider the feedback to be true, this is working on my local:

@Test
    fun failsIfFollowItself() {
        val accountInfo1 = AccountInfo("email@email.com", "janedoe", "31415926")
        val mockUser = RegisteredUser(accountInfo1, setOf())
        `when`(jwtService.getUserFromAuthorizationHeader("authorizationHeader")).thenReturn(mockUser)
        assertThrows(ResponseStatusException::class.java) {
            userInteractionController.followUser(
                "janedoe",
                "someAuthorization"
            )
        }
    }

Reply:
You are right, thanks for pointing out. Changed in the latest commit.